gremlin-shaded/pom.xml

```diff
@@ -24,6 +24,67 @@ limitations under the License.
     </parent>
     <artifactId>gremlin-shaded</artifactId>
     <name>Apache TinkerPop :: Gremlin Shaded</name>
+    <packaging>bundle</packaging>
+    <properties>
+        <!--
+        As this jar is shaded, the maven-bundle-plugin is not able to process exported and imported packages from generated classes.
+        So we must explicitely tell which packages are exported and imported by this bundle to the maven-bundle-plugin.
+        -->
+        <osgi.import>
```

Comment:
I see. That solves the typical trouble that OSGI solves. In OSGI you can load several versions of the same third library and each bundle tells which version it wants to use.
I understand your concern about this huge list of packages to maintain. I thought that the unit test will detect and tell if some packages must be added to the <osgi.import> and <osgi.export>.
But I'm struggling withe maven-shade-plugin and maven-bundle-plugin to make this bundle automatically without having to specify the list of packages to import and to export. I'm trying what's suggested in this [ticket](https://issues.apache.org/jira/browse/FELIX-1184) without any success. I don't give up...